### PR TITLE
better protocol fee tests

### PR DESCRIPTION
**Pull request by smarx**
## Description

These tests cover more of protocol fees. In particular, they add negative test cases around authorization and errors in fee collection. They also test more scenarios around initialized and uninitialized fee collectors.

## Testing instructions

`yarn build && yarn test` from `contracts/zero-ex`

## Types of changes

* more tests

## Checklist:

-   [X] Prefix PR title with `[WIP]` if necessary.
-   [X] Add tests to cover changes as needed.
-   [X] Update documentation as needed.
-   [X] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/test/protocol_fees_test.ts [hunk 1]
```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
...
+            const feeCollector = new FeeCollectorContract(
+                await protocolFees.getFeeCollector(pool).callAsync(),
+                env.provider,
+                env.txDefaults,
+            );
+
+            const tx = feeCollector
+                .initialize(weth.address, staking.address, pool)
+                .sendTransactionAsync({ from: unauthorized });
+            expect(tx).to.revertWith(new AuthorizableRevertErrors.SenderNotAuthorizedError(unauthorized));
```
Comment (mzhu25) [inline contracts/zero-ex/test/protocol_fees_test.ts:71]:
```suggestion
            return expect(tx).to.revertWith(new AuthorizableRevertErrors.SenderNotAuthorizedError(unauthorized));
```
`expect(tx).to.reverWith(...)` is async, so we should either await or return the promise. Our convention is to `return`.

File: contracts/zero-ex/test/protocol_fees_test.ts [hunk 2]
```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
...
-            // We leave 1 wei behind, of both ETH and WETH, for gas reasons.
-            const total = amount1.plus(amount2).minus(2);
-            return expect(balance).to.bignumber.eq(total);
+        const pool1 = hexUtils.random();
+        const pool2 = hexUtils.random();
+
+        it('should revert if WETH transfer fails', async () => {
+            const tooMuch = constants.ONE_ETHER.plus(1);
+            const tx = collect(pool1, constants.ONE_ETHER.plus(1), constants.ZERO_AMOUNT);
+            expect(tx).to.revertWith(
```
Comment (mzhu25) [inline contracts/zero-ex/test/protocol_fees_test.ts:82]: same here

File: contracts/zero-ex/test/protocol_fees_test.ts [hunk 3]
```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
...
+                    tooMuch,
+                    undefined,
+                ),
+            );
+        });
+
+        it('should revert if insufficient ETH transferred', async () => {
+            const tooLittle = constants.ONE_ETHER.minus(1);
+            const tx = collect(pool1, constants.ONE_ETHER, tooLittle);
+            expect(tx).to.revertWith('FixinProtocolFees/ETHER_TRANSFER_FALIED');
```
Comment (mzhu25) [inline contracts/zero-ex/test/protocol_fees_test.ts:96]: and here

File: contracts/zero-ex/test/protocol_fees_test.ts [hunk 4]
```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
         await weth.approve(protocolFees.address, constants.ONE_ETHER).awaitTransactionSuccessAsync({ from: payer });
     });
 
+    function collect(
+        poolId: string,
+        amount: BigNumber,
+        etherValue: BigNumber,
+    ): PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs> {
```
Comment (mzhu25) [inline contracts/zero-ex/test/protocol_fees_test.ts:45]: nit:
```suggestion
    async function collectAsync(
        poolId: string,
        amount: BigNumber,
        etherValue: BigNumber,
    ): Promise<TransactionReceiptWithDecodedLogs> {
```

File: contracts/zero-ex/test/protocol_fees_test.ts [hunk 5]
```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
...
+        return protocolFees
+            .collectProtocolFee(poolId, amount, weth.address)
+            .awaitTransactionSuccessAsync({ from: payer, value: etherValue });
+    }
+
+    function transferFees(poolId: string): PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs> {
```
Comment (mzhu25) [inline contracts/zero-ex/test/protocol_fees_test.ts:51]:
```suggestion
    async function transferFeesAsync(poolId: string): Promise<TransactionReceiptWithDecodedLogs> {
```

**Review by dorothy-zbornak — APPROVED** [review]
Looks good after addressing Michael's comments!

**Comment by smarx** [comment]
Thank you both for the review! @mzhu25, my latest changes should address all your feedback.

**Review by dorothy-zbornak — APPROVED** [review]

**Review by mzhu25 — APPROVED** [review]

**Pull request merged**